### DISCUSSION TITLE
Google authentication giving invalid credentials.

### DISCUSSION THREAD

**Discussion by abhinavgautam07** [provectus/kafka-ui#3183] [Q&A] [answered]
We have 2 setups of kafka, one each for dev and prod. I want to setup UI  with RBAC for both of them. Google authentication seems to be working fine on Dev, but giving invalid creds on prod.

config for dev and prod.
[configs.zip](https://github.com/provectus/kafka-ui/files/10325320/configs.zip)

**DEVELOPMENT**

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/49569646/210087144-6936dd9c-abb0-4f9f-8cc5-fb178548d587.png">

**PRODUCTION**
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/49569646/210087075-f8cad6e3-2612-493f-b2d9-6d74d3d161d8.png">

**Comment by Haarolean** [comment]
Hey, please try `redirect-uri` with dash instead of an underscore. Let me know how it goes.

Also, a tip: you can enable browser's developer console once you authenticate, you might get useful info upon an auth error (look at the URLs of requests).

**Reply by abhinavgautam07** [reply]
still the same issue.

Getting 302 not found in network call.

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/49569646/210498222-b64c652f-f962-4b0f-92ea-797d12dd08d2.png">

**Reply by Haarolean** [reply, marked as answer]
@abhinavgautam07 302 is "found", not "not found", it's a redirect. 
- Check the next request (login?error) is there anything useful there?
- Try removing redirect uri from your config at all
- Try adding a scope (`scope: openid`) along other variables.

**Comment by fedeostrit** [comment]
Hi, I wanted to know if there is any way to configure google's oauth2 and use the google provider to do RBAC by groups and not just by user or entire domain as shown in the documentation:
oauth2:

kafka:
  clusters:
    - name: local
      bootstrapServers: localhost:9092
    # ...

auth:
  type: OAUTH2
  oauth2:
    client:
      google:
        provider: google
        clientId: xxx.apps.googleusercontent.com
        clientSecret: GOCSPX-xxx
        user-name-attribute: email
        custom-params:
          type: google
          allowedDomain: provectus.com

provider:
    - name: "memelords"
      subjects:
        - provider: oauth_google
          type: domain
          value: "memelord.lol"


It only allows you to use domain or user, but if I have 100 users, I would have to create a role for each one and map it in a huge .yaml file. Also, when a new resource enters, it must be modified to add it.

**Reply by Haarolean** [reply]
https://github.com/provectus/kafka-ui/discussions/3274
